client/components/DisplayTest/index.jsx

```diff
@@ -46,6 +50,19 @@ class DisplayTest extends Component {
         this.performButtonAction();
     }
 
+    handleRedoClick() {
+        const { deleteResultFromTest, test } = this.props;
+
+        if (!test.result || test.result.status !== 'complete') {
+            document
+                .getElementById('test-iframe')
+                .contentWindow.location.reload();
```

Comment:
If, for any reason, this iframe is not in the dom, then this is going to blow up.